documentation/creating_a_script_plan.md

```diff
@@ -0,0 +1,369 @@
+```
+
+## Scripts with arguments
+
+### As environment variables
+
+You can pass arguments from your plan to your script as environment variables, which can be useful
+for scripts with optional parameters (where ordering arguments can be hard to predict), complex
+arguments, or structured data.
+
+Modify the example script above to take a command-line argument for the directory to run
+`docker-compose` from:
+
+_update\_images.sh_
+```
+#!/bin/sh
+if [ -n $DOCKER_COMPOSE_DIRECTORY ]
+  cd $DOCKER_COMPOSE_DIRECTORY
+fi
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+Run this script directly with Bolt by passing the `--env-var` command-line option to set
+environment variables during script execution:
+
+_\*nix shell command_
+```
+bolt script run manage_docker/scripts/update_images.sh -t <targets> --env-var DOCKER_COMPOSE_DIRECTORY=./docker
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltScript -Name manage_docker/scripts/update_images.sh -Targets <targets> -EnvVar DOCKER_COMPOSE_DIRECTORY=.\docker
+```
+
+Before you pass the argument through the plan as an environment variable,
+[add a plan parameter](writing_yaml_plans.md#parameters) for the argument and pass it through to the
+script. In a YAML plan, that should look something like this:
+```yaml
+parameters:
+  directory:
+    type: Optional[String]
+    default: .
+    description: The directory to execute 'docker-compose' from
+  targets:
+    ...
+```
+
+Next set that argument as an environment variable as part of the [script
+step](writing_yaml_plans.md#script_step)
+```yaml
+steps:
+  - name: run_script
+    script: manage_docker/scripts/update_images.sh
+    env_vars:
+      DOCKER_COMPOSE_DIRECTORY: ./docker
+    targets: $targets
+```
+
+And now your plan should look something like this:
+```yaml
+description: Update Docker images
+parameters:
+  directory:
+    type: Optional[String]
+    default: .
+    description: The directory to execute 'docker-compose' from
+  targets:
+    type: TargetSpec
+    description: A list of targets to run actions on
+
+steps:
+  - name: run_script
+    script: manage_docker/scripts/update_images.sh
+    env_vars:
+      DOCKER_COMPOSE_DIRECTORY: ./docker
+    targets: $targets
+
+return: $run_script
+```
+
+You can run the plan with:
+
+_\*nix shell command_
+```
+bolt plan run manage_docker::update_images -t <targets> directory=./docker
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltPlan -Name manage_docker::update_images -Targets <targets> directory=./docker
+```
+
+### As command-line arguments
+
+Passing script arguments on the command-line through Bolt is useful for scripts with simple,
+required arguments, like one or more required strings.
+
+For example, if your script takes a command-line argument for the directory instead of an environment variable:
```

Comment:
```suggestion
For example, the following script takes a command-line argument for the directory instead of an environment variable:
```

This sentence has a conditional clause, but the sentence after the script doesn't have a consequence, so it reads a bit weird.